test/test_prefix_setting/BUILD.bazel

```diff
@@ -0,0 +1,41 @@
+load("//rust:defs.bzl", "rust_test")
+load(":defs.bzl", "with_prefix_test_output_files_cfg")
+
+# Checks that rust_test targets can optionally have their output files
+# declared without a hash prefix. The prefix exists to mitigate filename
+# conflicts on non-sandboxed hosts, but may be unwanted when an existing
+# codebase with filename assumptions is being migrated to Rust.
+
+rust_test(
+    name = "test_binary",
+    srcs = ["lib.rs"],
+    edition = "2018",
+    tags = ["manual"],
+)
+
+with_prefix_test_output_files_cfg(
+    name = "test_binary_with_output_hash",
+    testonly = True,
+    srcs = [":test_binary"],
+    prefix_test_output_files = True,
+)
+
+with_prefix_test_output_files_cfg(
+    name = "test_binary_without_output_hash",
+    testonly = True,
+    srcs = [":test_binary"],
+    prefix_test_output_files = False,
+)
+
+sh_test(
```

Comment:
Do you need `sh_test`? Would it be possible to do this test with the bazel unittest framework? You're just trying to assert on the output path, right?

Reply:
Just lack of familiarity with the skylib testing system. Updated to replace sh_test with `unittest.bzl%analysistest`.